sonar-core/src/main/resources/org/sonar/core/component/db/ComponentMapper.xml

```diff
@@ -156,7 +156,7 @@
     </where>
   </select>
 
-  <select id="selectAllViewsAndSubViews" resultType="hashmap">
+  <select id="selectAllViewsAndSubViews" resultType="UuidWithProjectUuid">
```

Comment:
Indeed, the generic option seems good to me. 

By the way, searching in the table projects with only filters on qualifiers should require an index on the projects.qualifier column, don't you think (There's an index on snapshots.qualifier) ?